sdk/helper/storagepacker/ids_test.go

```diff
@@ -0,0 +1,54 @@
+package storagepacker
+
+import (
+	"context"
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/sdk/logical"
+	"testing"
+)
+
+func getStoragePackerBits(tb testing.TB, baseBucketBits int, bucketShardBits int) *StoragePackerV2 {
+	storage := &logical.InmemStorage{}
+	storageView := logical.NewStorageView(storage, "packer/buckets/v2")
+	storagePacker, err := NewStoragePackerV2(context.Background(), &Config{
+		BucketStorageView: storageView,
+		ConfigStorageView: logical.NewStorageView(storage, "packer/config"),
+		Logger:            log.New(&log.LoggerOptions{Name: "storagepackertest"}),
+		BaseBucketBits:    baseBucketBits,
+		BucketShardBits:   bucketShardBits,
+	})
+	if err != nil {
+		tb.Fatal(err)
+	}
+	return storagePacker.(*StoragePackerV2)
+}
+
+func TestStoragePackerV2_FirstBucketKey(t *testing.T) {
+	// Default settings
+	s_8_4 := getStoragePackerBits(t, 8, 4)
+
+	key, err := s_8_4.firstKey("12345678")
```

Comment:
Not sure where the values going into this algorithm originate, so this is probably moot. But would we ever need to worry about a string being inputted here that was a non-English character set? And also, would we ever need to worry about storage packer bits being like 3, instead of something cleanly divisible by 4?

Reply:
The config is checked for divisibility by 4; TestStoragePackerV2_CreationErrors checks that test.

It may well be that the IDs are arbitrary Unicode, but they are all hashed before being converted to keys, so keys are strictly hexadecimal strings. Do you think it's worth adding a test to check ID corner cases--- empty string, multi-byte chars, etc?

Reply:
That sounds safe to me! Also, as I continued reviewing the code, I found the spot where that happens. Looks good!